Bluetooth: L2CAP: Document behavior of alloc_buf

Vudentz opened the pull request:
Channel that requires segmentation must set alloc_buf in order for the
stack to be able to reassemble segments of an SDU.

If this happens at runtime warn the user and truncate the MTU to
force segmentation to be disabled.

Fixed #28384

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

joerchan commented on subsys/bluetooth/host/l2cap.c:
```diff
@@ -778,6 +778,19 @@ static void l2cap_chan_rx_init(struct bt_l2cap_le_chan *chan)
+	 */
+	chan->rx.mps = MIN(chan->rx.mtu + 2, L2CAP_MAX_LE_MPS);
+
+	/* Assert if channel have disabled segmentation but still have set an
```
The comment does not match the code, needs to be updated.

jhedberg reviewed (changes requested):
The changes look fine, but the commit message summary is rather misleading: when I read it I assume that there are only documentation changes involved, however when I look at the actual changes I see you're changing code as well. Could you change the commit message to something more accurate?

jhedberg commented:
..or alternatively split this into two commits: one that updates alloc_buf documentation and another which does the code change

Vudentz committed:
Bluetooth: L2CAP: Document behavior of alloc_buf

Channel that requires segmentation must set alloc_buf in order for the
stack to be able to reassemble segments of an SDU.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: L2CAP: Truncate RX MTU if segmentation is not supported

If hannel don't have alloc_buf and the RX MTU is configured to require
segmentation this warn the user and truncate the RX MTU.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz commented:
> ..or alternatively split this into two commits: one that updates alloc_buf documentation and another which does the code change

Done.
